src/core/Dropdown/index.tsx

```diff
@@ -100,87 +101,133 @@ export default function Dropdown<Multiple extends boolean | undefined = false>({
     }
   }, [propValue]);
 
-  const open = Boolean(anchorEl);
+  const [open, setOpen] = useState(false);
 
   return (
-    <>
-      <InputDropdownComponent
-        label={label}
-        onClick={handleClick}
-        sdsStage={open ? "userInput" : "default"}
-        {...InputDropdownProps}
-        {...rest}
-      />
-      <DropdownMenu
-        anchorEl={anchorEl}
-        open={open}
-        search={search}
-        onClose={handleClose}
-        multiple={multiple as Multiple}
-        value={
-          (multiple ? pendingValue : value) as MUIValue<
+    <ClickAwayListener onClickAway={handleClickAway}>
+      <div>
+        <InputDropdownComponent
+          label={label}
+          onClick={handleClick}
+          sdsStage={open ? "userInput" : "default"}
+          {...InputDropdownProps}
+          {...rest}
+        />
+        <DropdownMenu
+          anchorEl={anchorEl}
+          open={open}
+          search={search}
+          onClose={handleClose}
+          multiple={multiple as Multiple}
+          value={
+            (multiple ? pendingValue : value) as MUIValue<
+              DefaultDropdownMenuOption,
+              Multiple,
+              undefined,
+              undefined
+            >
+          }
+          onChange={handleChange}
+          disableCloseOnSelect={multiple}
+          PopperComponent={PopperComponent}
+          PopperBaseProps={{ sx: { minWidth: 250 } }}
+          options={options}
+          {...DropdownMenuProps}
+        >
+          {buttons ? (
+            <div>
+              {buttonPosition === "left" ? (
+                <div>
+                  <StyledButton
+                    onClick={handleClose}
+                    sdsStyle="square"
+                    sdsType="primary"
+                  >
+                    Apply
+                  </StyledButton>
+                  <StyledButton
+                    onClick={handleCancel}
+                    sdsStyle="square"
+                    sdsType="secondary"
+                  >
+                    Cancel
+                  </StyledButton>
+                </div>
+              ) : (
+                <div>
+                  <StyledButton
+                    onClick={handleCancel}
+                    sdsStyle="square"
+                    sdsType="secondary"
+                  >
+                    Cancel
+                  </StyledButton>
+                  <StyledButton
+                    onClick={handleClose}
+                    sdsStyle="square"
+                    sdsType="primary"
+                  >
+                    Apply
+                  </StyledButton>
+                </div>
+              )}
+            </div>
+          ) : null}
+        </DropdownMenu>
+      </div>
+    </ClickAwayListener>
+  );
+
+  function handleClickAway() {
+    if (open) {
+      setOpen(false);
+
+      if (multiple) {
+        setValue(
+          pendingValue as MUIValue<
```

Comment:
I see this type copy/pasted a few times. Might want to define it somewhere above and give it a shorter name.

Reply:
That makes sense 👍🏻 I'll update the repo.